lib/Mojolicious/Plugin/AssetPack/Pipe/OptiPng.pm

```diff
@@ -0,0 +1,94 @@
+
+package Mojolicious::Plugin::AssetPack::Pipe::OptiPng;
+
+use Mojo::Base 'Mojolicious::Plugin::AssetPack::Pipe';
+use Mojo::Util qw( spurt slurp );
+use File::Temp qw( tempfile );
+
+sub process {
+	my ($self, $assets) = @_;
+	my $store = $self->assetpack->store;
+	my $file;
+
+	$assets->each(
+		sub {
+			my ($asset, $index) = @_;
+
+			# might also use optipng to convert BMP, GIF, PNM or TIFF to PNG
+			# but not yet
+			return unless $asset->format eq 'png';
+
+			my $attrs = $asset->TO_JSON;
+			$attrs->{key}    = 'optipng';
+			$attrs->{format} = 'png';
+
+			# already processed before?
+			return $asset->content($file)->FROM_JSON($attrs) if $file = $store->load($attrs);
+
+			my ( $content, $error );
+			{
+				my ( $fh, $file ) = tempfile();
+				spurt $store->load($asset), $file;
+
+				my ( $in, $out, $err ) = ( '', '', '' );
+				my @cmd = ( 'optipng', '-quiet', '-clobber', $file );
```

Comment:
Yeah, "robust" was a pretty bad choice of word. Sorry about that. What I meant is that I consider the logic in ```_find_app()``` an enhancement. I also don't see what your code does that isn't already implemented in ```run()```.

https://github.com/jhthorsen/mojolicious-plugin-assetpack/blob/master/lib/Mojolicious/Plugin/AssetPack/Pipe.pm#L31